### PR TITLE
Speaking in sign language no longer reveals your identity 

### PR DESCRIPTION
**Pull request by epochayur**
## About The Pull Request

Added an argument to compose_message() (visible_name) that sets the name of the speaker to their visible identity, for messages that can be seen but not heard, and updated Hear() for sign language accordingly. 

## Why It's Good For The Game

Before, your name was attached to your sign message as if you were using your real voice. Now, your name on sign messages is based off your visual identity. 
It didn't make sense that signing revealed who you were even if you were wearing a disguise. This appeared to be an oversight rather than an intentional design decision. 

## Changelog

:cl: Epoc
fix: Speaking in sign language no longer reveals your identity 
/:cl:

### CODE REVIEW

**Comment by BlueMemesauce** [comment]
Now all the powergamers will only speak in sign language

**Comment by epochayur** [comment]
Do I have to say this?
No, using sign language does not work over the radio.

**Comment by BlueMemesauce** [comment]
You can sign with the translation gloves, which everyone that knows sign language are automatically given
image
But it seems that they don't allow you to remain unknown over the radio so I guess it isn't as bad as I thought

**Comment by BlueMemesauce** [comment]
> Sadly we can't patch out players being stupid, but if you see the captain walking around with a gasmask on while signing, consider why.

I think you misunderstood what I meant, the captain wouldn't hide his voice because he has no need. It's more beneficial that other people know who he is since they will most likely protect him.

I meant more like people who are antags/greytiders wanted by security that need to remain unknown. Sign language will allow them to hide their identity easier and remain unknown even while talking. That's why powergamers would take the perk since it would provide alot of value. 

Also, not being able to talk with your hands full isn't that much of a downside. You can simply enable/disable the perk whenever you like.

**Comment by epochayur** [comment]
> I think you misunderstood what I meant, the captain wouldn't hide his voice because he has no need. It's more beneficial that other people know who he is since they will most likely protect him.

I was more imagining a scenario where someone is masquerading as the captain and trying to keep their real identity a secret. To that point:

> Also, not being able to talk with your hands full isn't that much of a downside. You can simply enable/disable the perk whenever you like.

Disguising as someone and actually doing their job (i.e. using objects in their hands whether it be tools, guns, etc.) simultaneously seems to me like it would be a pain, and should naturally draw suspicion.

> I meant more like people who are antags/greytiders wanted by security that need to remain unknown. Sign language will allow them to hide their identity easier and remain unknown even while talking. That's why powergamers would take the perk since it would provide alot of value for currently only 4 points.

I do see your point. My hope would be that players would become aware of this strategy and treat gasmasked signers with greater suspicion. The purpose of this PR isn't balance changes, but more to address what I see as an oversight and make the game somewhat more realistic. 

If a maintainer would like me to change its point cost accordingly I can, but again, the point of this PR isn't balance. I wouldn't want to affect anyone's quirk build whose points are carefully balanced.

**Comment by BlueMemesauce** [comment]
Oh I understand what you mean with the Captain, I was the one who wasn't understanding it in that case.

Normally it would be considered suspicious, but that's the same way gas masks were balanced. It stopped being suspicious when every single assistant started wearing gas masks. 
But you're right, we shouldn't balance the game just because of people abusing the feature.  Maybe we should just see how it goes first and balance the points later, if all the assistants start speaking sign language.

File: code/game/say.dm [hunk 1]
```diff
@@ -117,6 +117,9 @@ GLOBAL_LIST_INIT(freqtospan, list(
 	if(face_name && ishuman(speaker))
 		var/mob/living/carbon/human/H = speaker
 		namepart = "[H.get_face_name()]" //So "fake" speaking like in hallucinations does not give the speaker away if disguised
+	else if(visible_name && ishuman(speaker))
+		var/mob/living/carbon/human/H = speaker
+		namepart = "[H.get_visible_name()]" //For if the message can be seen but not heard, shows "speaker"'s visible identity (like when using sign language)
```
Comment (san7890) [inline code/game/say.dm:122]:
```suggestion
		var/mob/living/carbon/human/human_speaker = speaker
		namepart = "[human_speaker.get_visible_name()]" //For if the message can be seen but not heard, shows "speaker"'s visible identity (like when using sign language)
```

no new single letter variables

**Review by san7890 — CHANGES_REQUESTED** [review]

**Pull request merged**